packages/create-plugin/src/commands/generate.command.ts

```diff
@@ -1,15 +1,186 @@
+  const commonActions = getActionsForTemplateFolder({
+    folderPath: TEMPLATE_PATHS.common,
+    exportPath,
+    templateData,
+  });
+
+  // Copy over files from the plugin type specific folder, e.g. "templates/app" for "app" plugins ("app" | "panel" | "datasource").
+  const pluginTypeSpecificActions = getActionsForTemplateFolder({
+    folderPath: TEMPLATE_PATHS[templateData.pluginType],
+    exportPath,
+    templateData,
+  });
+
+  // Copy over backend-specific files (if selected)
+  const backendFolderPath = templateData.isAppType ? TEMPLATE_PATHS.backendApp : TEMPLATE_PATHS.backend;
+  const backendActions = templateData.hasBackend
+    ? getActionsForTemplateFolder({ folderPath: backendFolderPath, exportPath, templateData })
+    : [];
+
+  // Common, pluginType and backend actions can contain different templates for the same destination.
+  // This filtering removes the duplicate file additions to make sure the correct template is scaffolded.
+  // Note that the order is reversed so backend > pluginType > common
+  const pluginActions = [...backendActions, ...pluginTypeSpecificActions, ...commonActions].reduce((acc, file) => {
+    const actionExists = acc.find((f) => f.path === file.path);
+    // return early to prevent duplicate file additions
+    if (actionExists) {
+      return acc;
+    }
+    acc.push(file);
+    return acc;
+  }, []);
+
+  // Copy over Github workflow files (if selected)
+  const ciWorkflowActions = templateData.hasGithubWorkflows
+    ? getActionsForTemplateFolder({
+        folderPath: TEMPLATE_PATHS.ciWorkflows,
+        exportPath,
+        templateData,
+      })
+    : [];
+
+  const isCompatibleWorkflowActions = templateData.hasGithubLevitateWorkflow
+    ? getActionsForTemplateFolder({
+        folderPath: TEMPLATE_PATHS.isCompatibleWorkflow,
+        exportPath,
+        templateData,
+      })
+    : [];
+
+  return [...pluginActions, ...ciWorkflowActions, ...isCompatibleWorkflowActions];
+}
+
+function getActionsForTemplateFolder({
+  folderPath,
+  exportPath,
+  templateData,
+}: {
+  folderPath: string;
+  exportPath: string;
+  templateData: TemplateData;
+}) {
+  let files = glob.sync(`${folderPath}/**`, { dot: true });
+
+  // The npmrc file is only useful for `pnpm` settings. We can remove it for other package managers.
+  if (templateData.packageManagerName !== 'pnpm') {
+    files = files.filter((file) => path.basename(file) !== 'npmrc');
+  }
+
+  function getFileExportPath(f: string) {
+    return path.relative(folderPath, path.dirname(f));
+  }
+
+  return files.filter(isFile).map((f) => ({
+    templateFile: f,
+    // The target path where the compiled template is saved to
+    path: path.join(exportPath, getFileExportPath(f), getExportFileName(f)),
+    data: {
+      ...EXTRA_TEMPLATE_VARIABLES,
+      ...templateData,
+    },
+  }));
+}
+
+async function generateFiles({ actions }: { actions: any[] }) {
```

Comment:
If I read this correctly. there's a chance the file generation fails and we end up with a half-way generated plugin. Was that also the case with plopjs?